src/main/groovy/com/bugsnag/android/gradle/BugsnagManifestTask.groovy

```diff
@@ -51,10 +51,20 @@ class BugsnagManifestTask extends BugsnagVariantOutputTask {
             application.appendNode('meta-data', [(ns.name): BugsnagPlugin.BUILD_UUID_TAG, (ns.value): buildUUID])
 
             // Write the manifest file
-            def writer = new FileWriter(manifestPath)
-            def printer = new XmlNodePrinter(new PrintWriter(writer))
-            printer.preserveWhitespace = true
-            printer.print(xml)
+            FileWriter writer = null
+
+            try {
+                writer = new FileWriter(manifestPath)
```

Comment:
can't you use try-with-resources here?

Reply:
This repo currently targets Java 1.6 at the moment, so unfortunately not without some changes.

I believe the reasoning behind targeting 1.6 is that we compile the local Gradle's distribution of Groovy, which as of [Gradle 4.0](https://docs.gradle.org/4.0/release-notes.html#groovy-upgraded-to-2.4.11) is Groovy 2.4.X. This version of Groovy still supports [1.6 JVMs](http://www.groovy-lang.org/download.html#requirements).

I'll raise an internal ticket to discuss bumping our compatibility to 1.7, as I'd hope most users of Android Studio are using a modern JDK.

Reply:
ah I see 👍